src/ripple/app/misc/NetworkOPs.cpp

```diff
@@ -1276,6 +1300,17 @@ NetworkOPsImp::doTransactionSync(
         transaction->setApplying();
     }
 
+    doTransactionSyncBatch(
+        lock, [&transaction](std::unique_lock<std::mutex>& lock) {
+            return transaction->getApplying();
```

Comment:
I think this was done intentionally because it avoids the overhead of taking a lock, and the consequences of a race condition are nearly-zero.

1. Most significantly, almost all accesses of the `*Applying` functions are under `NetworkOPsImp`'s lock. The exceptions are
    1. The new `processTransactionSet` function.
    2. The `popAcctTransaction` loop we were just talking about.
3. If there is a race condition, and `getApplying` returns false when it should be true, the transaction will be processed again. Not that big a deal if it's a rare one-off. Most of the time, it'll get `tefALREADY` or `tefPAST_SEQ`.
4. On the flip side, if it returns true, when it should be false, then the transaction must have been attempted recently, so no big deal if it doesn't immediately get tried right away.
5. If there's a race between `setApplying` and `clearApplying`, again, the flag ends up set, then a batch is about to try to process the transaction and will call `clearApplying` later. If it ends up cleared, then it might get attempted again later as is the case with item 2.

Alllll that said, I don't think it would hurt to rewrite those two lock exceptions I pointed out to ensure that the lock is taken for the function calls.